shared/bots/__init__.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from typing import List, Optional
+
+from shared.bots.github_apps import get_github_app_info_for_owner
+from shared.bots.owner_bots import get_owner_appropriate_bot_token
+from shared.bots.public_bots import get_token_type_mapping
+from shared.bots.repo_bots import get_repo_appropriate_bot_token
+from shared.bots.types import (
+    AdapterAuthInformation,
+)
+from shared.django_apps.codecov_auth.models import (
+    GITHUB_APP_INSTALLATION_DEFAULT_NAME,
+    Owner,
+    Service,
+)
+from shared.django_apps.core.models import Repository
+from shared.typings.torngit import GithubInstallationInfo
+
+log = logging.getLogger(__name__)
+
+
+def get_adapter_auth_information(
+    owner: Owner,
+    repository: Optional[Repository] = None,
+    *,
+    ignore_installations: bool = False,
+    installation_name_to_use: str | None = GITHUB_APP_INSTALLATION_DEFAULT_NAME,
+) -> AdapterAuthInformation:
+    """
+    Gets all the auth information needed to send requests to the provider.
+    This is mostly used by worker and api should use the token provided by the person making the request
```

Comment:
[nit] Suggestion for comment (should have suggested earlier, my bad)
```python
    Gets all the auth information needed to send requests to the provider.
    This logic is used by the worker to get the data needed to create a torngit.BaseAdapter
    :warning: Api should use the `owner.oauth_token` of the user making the request
```

Reply:
Np! Thanks for the propmpt